### ISSUE TITLE
If there are more than 2000 rooms, xmpp client setup never completes.

### ISSUE THREAD

**Issue by woparry** [skoczen/will#234]
Setting debug logging on shows this:

```
INFO     Got 1000 rooms
INFO     Starting new HTTPS connection (1): api.hipchat.com
DEBUG    "GET /v2/room?auth_token=****&expand=items&start-index=1000&max-results=1000 HTTP/1.1" 200 None
INFO     Got 1000 rooms
INFO     Starting new HTTPS connection (1): api.hipchat.com
DEBUG    "GET /v2/room?auth_token=****&expand=items&start-index=1000&max-results=1000 HTTP/1.1" 200 None
INFO     Got 1000 rooms
INFO     Starting new HTTPS connection (1): api.hipchat.com
DEBUG    "GET /v2/room?auth_token=****&expand=items&start-index=1000&max-results=1000 HTTP/1.1" 200 151147
```

we're setting start-index to 1000 every time instead of incrementing it by 1000 every time.

see will/mixins/room.py:

```
                if len(rooms['items']) == max_results:
                    params['start-index'] = max_results
                else:
                    break
```

**Comment by sarpk** [comment]
use V1_TOKEN

**Comment by woparry** [comment]
V1_TOKEN is not an adequate replacement for several reasons:
  - V1 api is deprecated
  - V1 tokens are much more difficult for users to get, since they must be generated by the group administrator (rather than being self-serve on the hipchat website)
  - hanging forever is NOT acceptable behavior if a V2 token is present instead of a V1 one. Being slower due to issuing multiple requests is.

**Comment by skoczen** [comment]
This should be fixed in 0.9.4 and up - please reopen if that's not the case!
